### ISSUE TITLE
"500 Internal Server Error" happens sometimes

### ISSUE THREAD

**Issue by david-ziegler** [kartevonmorgen/openfairdb#125]
it sometimes happens (a few times per week) that it says "Server currently not available" in the kartevonmorgen.org-Frontend which is triggered by an error-response for the `search`-request. I can't reproduce it right now so I can't say what the response is exactly. 
Helmut once tried the csv-export when this happened and he got: 
```
500: Internal Server Error
The server encountered an internal error while processing this request.
Rocket
```
This was on November 22 between 13:20 and 13:40. Do the logs show something?

What could that be? Is it possible that the whole openfairdb crashes in that moment? Do we restart the service automatically?

**Comment by uklotzde** [comment]
Many database errors:
```
Nov 22 13:22:05 flosse openfairdb_stable-start[14347]: ERROR 2018-11-22T12:22:05Z: openfairdb::ports::web::api: Error: database is locked
```
Caused by concurrently started backup script(s) stuck in retry loop or by too many concurrent connections used by the service.

**Comment by uklotzde** [comment]
``` 1384 ?        S      0:00  \_ /nix/store/kvhap1km6mq5bi5mzlhkqr9sgdllhycq-cron-4.1/bin/cron -n
 1385 ?        Ss     0:00  |   \_ /nix/store/zqh3l3lyw32q1ayb15bnvg9f24j5v2p0-bash-4.4-p12/bin/bash -c /etc/nixos/ofdb/backup-sqlite.sh > /dev/null 2>> /var/db/openfairdb/data/backup/error.
 1386 ?        S     63:09  |       \_ /bin/sh /etc/nixos/ofdb/backup-sqlite.sh
11286 ?        S      0:00  |           \_ sleep 1
11689 ?        S      0:00  \_ /nix/store/kvhap1km6mq5bi5mzlhkqr9sgdllhycq-cron-4.1/bin/cron -n
11690 ?        Ss     0:00  |   \_ /nix/store/zqh3l3lyw32q1ayb15bnvg9f24j5v2p0-bash-4.4-p12/bin/bash -c /etc/nixos/ofdb/backup-sqlite.sh > /dev/null 2>> /var/db/openfairdb/data/backup/error.
11691 ?        S     61:07  |       \_ /bin/sh /etc/nixos/ofdb/backup-sqlite.sh
11274 ?        S      0:00  |           \_ sleep 1
31350 ?        S      0:00  \_ /nix/store/kvhap1km6mq5bi5mzlhkqr9sgdllhycq-cron-4.1/bin/cron -n
31351 ?        Ss     0:00  |   \_ /nix/store/zqh3l3lyw32q1ayb15bnvg9f24j5v2p0-bash-4.4-p12/bin/bash -c /etc/nixos/ofdb/backup-sqlite.sh > /dev/null 2>> /var/db/openfairdb/data/backup/error.
31352 ?        S     60:39  |       \_ /bin/sh /etc/nixos/ofdb/backup-sqlite.sh
11258 ?        S      0:00  |           \_ sleep 1
27903 ?        S      0:00  \_ /nix/store/kvhap1km6mq5bi5mzlhkqr9sgdllhycq-cron-4.1/bin/cron -n
27904 ?        Ss     0:00  |   \_ /nix/store/zqh3l3lyw32q1ayb15bnvg9f24j5v2p0-bash-4.4-p12/bin/bash -c /etc/nixos/ofdb/backup-sqlite.sh > /dev/null 2>> /var/db/openfairdb/data/backup/error.
27905 ?        S     59:50  |       \_ /bin/sh /etc/nixos/ofdb/backup-sqlite.sh
11241 ?        S      0:00  |           \_ sleep 1
 1691 ?        S      0:00  \_ /nix/store/kvhap1km6mq5bi5mzlhkqr9sgdllhycq-cron-4.1/bin/cron -n
 1692 ?        Ss     0:00  |   \_ /nix/store/zqh3l3lyw32q1ayb15bnvg9f24j5v2p0-bash-4.4-p12/bin/bash -c /etc/nixos/ofdb/backup-sqlite.sh > /dev/null 2>> /var/db/openfairdb/data/backup/error.
 1693 ?        S     59:09  |       \_ /bin/sh /etc/nixos/ofdb/backup-sqlite.sh
11300 ?        S      0:00  |           \_ sleep 1
```
...and many more...

**Comment by flosse** [comment]
... so this is related to #70

**Comment by uklotzde** [comment]
Should be fixed by #148 unless external processes are accessing the same database.
